Data race detected in TestHealthCheckExtensionUsage

### Component(s)

_No response_

### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4858076079/jobs/8659170433?pr=21399

```
WARNING: DATA RACE
Read at 0x00c00008c388 by goroutine 102:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExporter).rotate()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/exporter.go:60 +0xb5
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2.1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:95 +0x7a

Previous write at 0x00c00008c388 by goroutine 100:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func11()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:210 +0x135
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func20()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:326 +0x7d4
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1[57](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4858076079/jobs/8659170433?pr=21399#step:8:58)6 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47

Goroutine 102 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:91 +0x264

Goroutine 100 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x805
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:310 +0x1785
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47
==================
==================
WARNING: DATA RACE
Read at 0x00c000094200 by goroutine 102:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExporter).rotate()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/exporter.go:63 +0x15e
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2.1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:95 +0x7a

Previous write at 0x00c000094200 by goroutine 100:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func11()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:210 +0xd4
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func20()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:326 +0x7d4
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1[62](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4858076079/jobs/8659170433?pr=21399#step:8:63)9 +0x47

Goroutine 102 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:91 +0x264

Goroutine 100 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x805
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:310 +0x1785
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47
==================
==================
WARNING: DATA RACE
Read at 0x00c00008b368 by goroutine 102:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExporter).rotate()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/exporter.go:64 +0x184
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2.1()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:95 +0x7a

Previous write at 0x00c00008b368 by goroutine 100:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.viewData()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:423 +0x144
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func11()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:210 +0x2e
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage.func20()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:326 +0x7d4
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x[64](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4858076079/jobs/8659170433?pr=21399#step:8:65)/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47

Goroutine 102 (running) created at:
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.(*healthCheckExtension).Start.func2()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension.go:91 +0x264

Goroutine 100 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x805
  github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension.TestHealthCheckExtensionUsage()
      /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/healthcheckextension/healthcheckextension_test.go:310 +0x1785
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47
==================
--- FAIL: TestHealthCheckExtensionUsage (8.03s)
    --- FAIL: TestHealthCheckExtensionUsage/WithCheckCollectorPipeline (1.00s)
        testing.go:[144](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4858076079/jobs/8659170433?pr=21399#step:8:145)6: race detected during execution of test
    testing.go:1446: race detected during execution of test
```

--- a/extension/healthcheckextension/healthcheckextension_test.go
+++ b/extension/healthcheckextension/healthcheckextension_test.go
@@ -196,6 +196,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},
@@ -207,6 +209,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},
@@ -233,6 +237,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},
@@ -244,6 +250,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},
@@ -272,6 +280,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},
@@ -285,6 +295,8 @@ func TestHealthCheckExtensionUsage(t *testing.T) {
 				},
 				{
 					step: func(hcExt *healthCheckExtension) error {
+						hcExt.exporter.mu.Lock()
+						defer hcExt.exporter.mu.Unlock()
 						hcExt.exporter.exporterFailureQueue = append(hcExt.exporter.exporterFailureQueue, viewData())
 						return hcExt.Ready()
 					},

